modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/ResizeHelper.java

```diff
@@ -0,0 +1,595 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.sun.javafx.scene.control;
+
+import java.util.BitSet;
+import java.util.List;
+import javafx.scene.control.ResizeFeaturesBase;
+import javafx.scene.control.TableColumnBase;
+
+/**
+ * Helps resize Tree/TableView columns.
+ * https://bugs.openjdk.org/browse/JDK-8293119
+ */
+public class ResizeHelper {
+    private static final int SMALL_DELTA = 32;
+    private final ResizeFeaturesBase rf;
+    private final int target;
+    private final List<? extends TableColumnBase<?,?>> columns;
+    private final int count;
+    private final ConstrainedColumnResize.ResizeMode mode;
+    private final int[] size;
+    private final int[] min;
+    private final int[] pref;
+    private final int[] max;
+    private final BitSet skip;
+
+    public ResizeHelper(ResizeFeaturesBase rf,
+                        double target,
+                        List<? extends TableColumnBase<?,?>> columns,
+                        ConstrainedColumnResize.ResizeMode mode) {
+        this.rf = rf;
+        this.target = (int)target;
+        this.columns = columns;
+        this.mode = mode;
+
+        this.count = columns.size();
+        size = new int[count];
+        min = new int[count];
+        pref = new int[count];
+        max = new int[count];
```

Comment:
Using integer arithmetic may not be the right thing to do here.

Reply:
You are right.  My initial desire to round came from TableUtil:260, where column sizes are rounded.  Not only this code is executed only once during application lifecycle, but also it is incorrect.  
Will revert back to doubles and remove rounding, while keeping track of possible rounding created by snapping.
Thank you.